src/config/mod.rs

```diff
@@ -123,23 +112,40 @@ fn load_asconfigc(wg_mod_home: &PathBuf) -> Result<ASConfigc, ConfigsError> {
 }
 
 fn load_settings(wg_mod_home: &PathBuf) -> Result<Settings, ConfigsError> {
-    let settings_file = wg_mod_home.join("settings.json");
+    let settings_manager = load_settings_manager(wg_mod_home);
 
-    if !settings_file.exists() {
-        create_default_settings_file(&settings_file)?
-    };
+    let mut settings = Settings::new(&settings_manager)
+        .map_err(|_| ConfigsError::SettingsError)?;
```

Comment:
Not good, you can't discard the error generated by Settings::new you need to store her in SettingsError.

In this state, the error trace of the program will be:
```
thread 'main' panicked at src/main.rs:11:21:
Unable to load wg-mod configs: SettingsError
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```